packages/atlas/src/components/_inputs/TextFieldWithDropdown/TextFieldWithDropdown.tsx

```diff
@@ -0,0 +1,105 @@
+import { useCombobox } from 'downshift'
+import React, { useEffect, useRef, useState } from 'react'
+
+import { ListItem, ListItemProps } from '@/components/ListItem'
+import { SvgActionLoader } from '@/components/_icons'
+
+import { ListWrapper, TextFieldWithDropdownWrapper } from './TextFieldWithDropdown.styles'
+
+import { TextField, TextFieldProps } from '../TextField'
+
+export type TextFieldWithDropdownProps<ItemType = unknown> = {
+  items?: (ListItemProps & ItemType)[]
+  loading?: boolean
+  onSelectedItemChange?: (item?: ListItemProps & ItemType) => void
+  onInputValueChange?: ((item?: string) => void) | ((item?: string) => Promise<void>)
+  resetOnSelect?: boolean
+  notFoundNode?: ListItemProps | null
+} & Omit<TextFieldProps, 'charactersCount'>
+
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-constraint
+export const TextFieldWithDropdown = <ItemType extends unknown>(props: TextFieldWithDropdownProps<ItemType>) => {
+  const {
+    loading,
+    items = [],
+    onSelectedItemChange,
+    onInputValueChange,
+    resetOnSelect,
+    notFoundNode,
+    error,
+    ...textFieldProps
+  } = props
+  const [inputItems, setInputItems] = useState<(ListItemProps & ItemType)[]>([])
+  const textFieldWithDropdownWrapperRef = useRef<HTMLDivElement>(null)
+  const textFieldRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (items) {
+      setInputItems(items)
+    }
+  }, [items])
+
+  const { isOpen, getMenuProps, getInputProps, highlightedIndex, getItemProps, getComboboxProps, reset } = useCombobox({
+    items: inputItems,
+    itemToString: (item) => (item ? item.label : ''),
+    onSelectedItemChange: ({ selectedItem }) => {
+      if (!selectedItem) {
+        return
+      }
+      onSelectedItemChange?.(selectedItem)
+      setInputItems([])
+      if (resetOnSelect) {
+        reset()
+      }
+    },
+    onInputValueChange: ({ inputValue }) => {
+      if (!inputValue) {
+        reset()
+        return
+      }
+      const filteredItems = items.filter((item) => item.label.toLowerCase().startsWith(inputValue?.toLowerCase()))
+      setInputItems(filteredItems)
+      onInputValueChange?.(inputValue)
+    },
+  })
+
+  const getTextFieldBottomEdgePosition = () => {
+    if (!textFieldRef.current || !textFieldWithDropdownWrapperRef.current) {
+      return
+    }
+    const { y: wrapperY } = textFieldWithDropdownWrapperRef.current.getBoundingClientRect()
+    const { y: inputY, height: inputHeight } = textFieldRef.current.getBoundingClientRect()
+    return inputY - wrapperY + inputHeight
+  }
```

Comment:
This will position the dropdown list below the input box. This is needed because if we add helper text to the text field, the list will be moved down below this helper text.
![image](https://user-images.githubusercontent.com/51168865/155946748-09e5dcd3-9ebd-4b76-b9ad-866fa1bb6f58.png)
I couldn't find a cleaner solution for this. I did a small refactoring and made sure that this will be not called when the helper text is not set. I provided a comment as well.